R/derive_var_trtemfl.R

```diff
@@ -0,0 +1,237 @@
+#' Derive Treatment-emergent Flag
+#'
+#' Derive treatment emergent analysis flag (e.g., `TRTEMFL`).
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by `start_date`, `end_date`, `trt_start_date`,
+#'   `trt_end_date`, `initial_intensity`, and `intensity` are expected.
+#'
+#' @param new_var New variable
+#'
+#' @param start_date Event start date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset
+#'
+#' @param end_date Event end date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset
+#'
+#' @param trt_start_date Treatment start date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset
+#'
+#' @param trt_end_date Treatment end date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset or `NULL`
+#'
+#' @param end_window
+#'
+#'    If the argument is specified, events starting more than the specified
+#'    number of days after end of treatment, are not flagged.
+#'
+#'    *Permitted Values:* A non-negative integer or `NULL`
+#'
+#' @param ignore_time_for_trt_end
+#'
+#'   If the argument is set to `TRUE`, the time part is ignored for checking if
+#'   the event occurred more than `end_window` days after end of treatment.
+#'
+#'   *Permitted Values:* `TRUE`, `FALSE`
+#'
+#' @param initial_intensity Initial severity/intensity or toxity
```

Comment:
@bundfussr sorry for late input but as i'm reviewing the admiralroche equivalent function thought i'd start here first to understand the admiral version. i spotted typos in few places for "toxity" -> "toxicity". not sure how spelling workflow check missed that one.